src/main/java/org/mockito/internal/junit/UniversalTestListener.java

```diff
@@ -34,6 +36,7 @@ public void testFinished(TestFinishedEvent event) {
         Collection<Object> createdMocks = mocks.keySet();
         //At this point, we don't need the mocks any more and we can mark all collected mocks for gc
         //TODO make it better, it's easy to forget to clean up mocks and we still create new instance of list that nobody will read, it's also duplicated
+        //TODO clean up all other state, null out stubbingLookupListener
```

Comment:
Your suggestion is not super clear to me. Given that this refactoring is not critical for the feature nor code quality, let's move on for now. We can get back to this if needed!